### DISCUSSION TITLE
Synchronization simulation or evaluation

### DISCUSSION THREAD

**Discussion by ahfeizi** [simnibs/simnibs#334] [2. General]
Can we evaluate or simulate synchronization of some brain regions after stimulation by SimNIBS?

**Comment by TorgeW** [comment]
That might be possible, but not with SimNIBS. 
SimNIBS can simulate time variant electric stimulation, for everything else, you would need to use another tool.

**Reply by 784622315** [reply]
Excuse me, can simNIBS perform time-varying magnetic stimulation? Can I input pulse waveform? How do you do this and can you finally get a time-dependent change in the electric field strength of a point? Thank you for your answer.

**Reply by axelthi** [reply]
Hi, as the system is quasi-static at the frequencies of TMS, it is sufficient to simulate the induced electric fields once for a chosen dI/dt (most easily chose 1 A/s). Afterwards, the induced electric field can be determined for any time point t during the TMS pulse by multiplying with the temporal derivative of the current pulse waveform ( i.e. E(1A/s)*dI/dt(t), whereby E is the precalculated electric field, and dI/dt is the derivative of the waveform)

**Reply by 784622315** [reply]
Thank you very much for your answer, your answer made me understand how to solve this problem, but I still have some questions, first of all, how to get the value of the electric field strength E, the second is how to simulate when the input current pulse is two high frequency sinusoids with frequency difference. I would appreciate it if you could answer my question

**Reply by bjornsig** [reply]
You can have a look at this example code for doing a simple temporal interference stimulation:

https://github.com/simnibs/simnibs/blob/master/simnibs/examples/simulations/TI.py
